### DISCUSSION TITLE
Redirecting factory constructors without default values

### DISCUSSION THREAD

**Discussion by kaboc** [rrousselGit/freezed#728] [Q&A]
```dart
@freezed
class IntList with _$IntList {
  const IntList._();

  const factory IntList({@Default([]) List<int> list}) = _IntList;

  const factory IntList.empty() = IntList;

  int? get first => list.isEmpty ? null : list.first;  // list cannot be accessed.
}
```

Running build_runner generates code with no errors nor warnings, but the generated abstract class does not have a getter for `list`.
It is resolved if some default value is set as below.

```dart
const factory IntList({@Default([]) List<int> list}) = _IntList;

const factory IntList.empty({@Default([]) List<int> list}) = IntList;
```

However, it is pointless to have two constructors with the same argument and default value. It does not make sense either that the constructor with the name "empty" can be given a value and become non-empty.

If Freezed is not used:

```dart
class IntList {
  const IntList({this.list = const []});

  const factory IntList.empty() = IntList;

  final List<int> list;

  int? get first => list.isEmpty ? null : list.first;
}
```

It seems like a bug if what is possible without Freezed is impossible with Freezed. Is it a known issue and documented anywhere? This was unexpected for me and it took me quite some time to realise the cause.

**Comment by rrousselGit** [comment]
The meaning of defining multiple constructors is different in Freezed

You defined a union type. But you clearly doesn't want a union

A plain classes likely makes more sense.

**Reply by kaboc** [reply]
If I want a constructor like the one I showed in a class that uses Freezed, how can I achieve it? A static function instead?

```dart
static IntList empty() => IntList();
```

This certainly works, but it cannot be used in the same way as a **const** constructor.
